### PR TITLE
Remove own dev directory management

### PR DESCRIPTION
**Pull request by balamurugana**
Signed-off-by: Bala.FA <bala@minio.io>

### CODE REVIEW

File: pkg/drive/drive.go [hunk 1]
```diff
@@ -18,40 +18,82 @@ package drive
...
+	}
+	return "/dev/" + name, nil
+}
+
+func mknod(fsuuid string, major, minor uint32) error {
+	path := filepath.Join(sys.DirectCSIDevRoot, fsuuid)
+
+	if err := os.Remove(path); err != nil && !errors.Is(err, os.ErrNotExist) {
+		return err
+	}
```
Comment (Praveenrajmani) [inline pkg/drive/drive.go:63]: question: will `os.Remove` throw a "device busy" if the device is mounted?

Comment (Praveenrajmani) [reply pkg/drive/drive.go:63]: can we mknod only if the (major, minor) of the device is different from what we expect? 

i see that, we call this function during drive formatting, in-case if the drive formatting fails due to any error, the listener will retry the format requests multiple times.

in such cases, we may delete and create the device files too many times.

**Review by Praveenrajmani — APPROVED** [review]
Few questions on `mknod`

LGTM otherwise

File: pkg/drive/drive.go (hunk 1 above)
Comment (balamurugana) [reply pkg/drive/drive.go]: 1. I haven't checked that. For regular files, `os.Remove()` succeeded even file is in use.
2. The better way is just do `Mknod()` and ignore `os.ErrExist` error.

**Review by wlan0 — DISMISSED** [review]

File: pkg/drive/drive.go [hunk 2]
```diff
@@ -18,40 +18,78 @@ package drive
...
+	name, err := sys.GetDeviceName(major, minor)
+	if err != nil {
+		return "", err
+	}
+	return "/dev/" + name, nil
+}
+
+func mknod(fsuuid string, major, minor uint32) error {
+	path := filepath.Join(sys.DirectCSIDevRoot, fsuuid)
+	err := unix.Mknod(path, unix.S_IFBLK|0666, int(unix.Mkdev(major, minor)))
```
Comment (Praveenrajmani) [inline pkg/drive/drive.go:60]: can we remove and create? to handle path changes? Like we had earlier

**Review by wlan0 — APPROVED** [review]

**Pull request merged**